test/containers.ts

```diff
@@ -0,0 +1,225 @@
+import { describe, expect, beforeEach, afterEach, vi, it } from "vitest";
+import { SelfClearingMap, SelfClearingSet } from "../src/utils/containers.js";
+
+import { setFlagsFromString } from "v8";
+import { runInNewContext } from "vm";
+
+/**
+ * Ensure we garbage collect the maps between tests, otherwise we might get false failures
+ */
+let gc: () => void = global.gc as unknown as () => void;
+if (!(gc as unknown as boolean)) {
+    setFlagsFromString("--expose_gc");
+    gc = runInNewContext("gc"); // nocommit
```

Comment:
That was a vestigial part of my snippet since it should never be committed to a running system, as argument changes after the VM started can cause a crash, but that's not a problem for unit tests, as you can just re-run them (and the failure rate is pretty low)